### ISSUE TITLE
training step refactor and testing

### ISSUE THREAD

**Issue by jaidhyani** [delphi-suite/delphi#76]
The actual training step should be factored into its own function and tested to check that gradient accumulation is working as expected.
